### PR TITLE
test for assinee model (issue-431)

### PR DESCRIPTION
**Pull request by probaku1234**
Pull request for [issue-431](https://github.com/mozilla/bugbug/issues/431)
implemented test function

### CODE REVIEW

File: tests/test_assignee.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from bugbug.models.assignee import AssigneeModel
+
+
+def test_get_assignee_labels():
+    # Due to the lack of data in fixtures/bugs.json, classes is always empty.
```
Comment (marco-c) [inline tests/test_assignee.py]: We need to add the required data to the fixture DB, otherwise the test is useless.

Comment (probaku1234) [reply tests/test_assignee.py]: Yes, I think so too. But I don't know how to add more data. Where are these bug data came from in bugs.json?

Comment (marco-c) [reply tests/test_assignee.py]: They come from data/bugs.json

**Comment by probaku1234** [comment]
I added test data and edit test function, but failed on bugbug tests in Taskcluster. I'm currently working on that.

**Comment by probaku1234** [comment]
I have no idea how to fix for failing test.
```
=================================== FAILURES ===================================
[31m[1m___________________________________ test_run ___________________________________[0m

[1m    def test_run():[0m
[1m        # Test running the training for the bug model.[0m
[1m>       run.main(run.parse_args(["--train", "--goal", "defect"]))[0m

[1m[31m/bugbug/tests/test_run.py[0m:11: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[1m[31m/bugbug/run.py[0m:90: in main
[1m    model.train()[0m
[1m[31m/bugbug/bugbug/model.py[0m:171: in train
[1m    y_test_filter, y_pred_filter, labels=class_names[0m
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

y_true = array([], dtype=float64), y_pred = array([], dtype=float64)
labels = array([1, 0]), sample_weight = None

[1m    def confusion_matrix(y_true, y_pred, labels=None, sample_weight=None):[0m
[1m        """Compute confusion matrix to evaluate the accuracy of a classification[0m
[1m    [0m
[1m        By definition a confusion matrix :math:`C` is such that :math:`C_{i, j}`[0m
[1m        is equal to the number of observations known to be in group :math:`i` but[0m
[1m        predicted to be in group :math:`j`.[0m
[1m    [0m
[1m        Thus in binary classification, the count of true negatives is[0m
[1m        :math:`C_{0,0}`, false negatives is :math:`C_{1,0}`, true positives is[0m
[1m        :math:`C_{1,1}` and false positives is :math:`C_{0,1}`.[0m
[1m    [0m
[1m        Read more in the :ref:`User Guide <confusion_matrix>`.[0m
[1m    [0m
[1m        Parameters[0m
[1m        ----------[0m
[1m        y_true : array, shape = [n_samples][0m
[1m            Ground truth (correct) target values.[0m
[1m    [0m
[1m        y_pred : array, shape = [n_samples][0m
[1m            Estimated targets as returned by a classifier.[0m
[1m    [0m
[1m        labels : array, shape = [n_classes], optional[0m
[1m            List of labels to index the matrix. This may be used to reorder[0m
[1m            or select a subset of labels.[0m
[1m            If none is given, those that appear at least once[0m
[1m            in ``y_true`` or ``y_pred`` are used in sorted order.[0m
[1m    [0m
[1m        sample_weight : array-like of shape = [n_samples], optional[0m
[1m            Sample weights.[0m
[1m    [0m
[1m        Returns[0m
[1m        -------[0m
[1m        C : array, shape = [n_classes, n_classes][0m
[1m            Confusion matrix[0m
[1m    [0m
[1m        References[0m
[1m        ----------[0m
[1m        .. [1] `Wikipedia entry for the Confusion matrix[0m
[1m               <https://en.wikipedia.org/wiki/Confusion_matrix>`_[0m
[1m               (Wikipedia and other references may use a different[0m
[1m               convention for axes)[0m
[1m    [0m
[1m        Examples[0m
[1m        --------[0m
[1m        >>> from sklearn.metrics import confusion_matrix[0m
[1m        >>> y_true = [2, 0, 2, 2, 0, 1][0m
[1m        >>> y_pred = [0, 0, 2, 2, 0, 2][0m
[1m        >>> confusion_matrix(y_true, y_pred)[0m
[1m        array([[2, 0, 0],[0m
[1m               [0, 0, 1],[0m
[1m               [1, 0, 2]])[0m
[1m    [0m
[1m        >>> y_true = ["cat", "ant", "cat", "cat", "ant", "bird"][0m
[1m        >>> y_pred = ["ant", "ant", "cat", "cat", "ant", "cat"][0m
[1m        >>> confusion_matrix(y_true, y_pred, labels=["ant", "bird", "cat"])[0m
[1m        array([[2, 0, 0],[0m
[1m               [0, 0, 1],[0m
[1m               [1, 0, 2]])[0m
[1m    [0m
[1m        In the binary case, we can extract true positives, etc as follows:[0m
[1m    [0m
[1m        >>> tn, fp, fn, tp = confusion_matrix([0, 1, 0, 1], [1, 1, 1, 0]).ravel()[0m
[1m        >>> (tn, fp, fn, tp)[0m
[1m        (0, 2, 1, 1)[0m
[1m    [0m
[1m        """[0m
[1m        y_type, y_true, y_pred = _check_targets(y_true, y_pred)[0m
[1m        if y_type not in ("binary", "multiclass"):[0m
[1m            raise ValueError("%s is not supported" % y_type)[0m
[1m    [0m
[1m        if labels is None:[0m
[1m            labels = unique_labels(y_true, y_pred)[0m
[1m        else:[0m
[1m            labels = np.asarray(labels)[0m
[1m            if np.all([l not in y_true for l in labels]):[0m
[1m>               raise ValueError("At least one label specified must be in y_true")[0m
[1m[31mE               ValueError: At least one label specified must be in y_true[0m

[1m[31m/usr/local/lib/python3.7/site-packages/sklearn/metrics/classification.py[0m:262: ValueError
```

File: tests/fixtures/bugs.json [hunk 2]
```diff
@@ -32,3 +32,9 @@
...
...\nbp-5a09146a-9cb4-40e3-b831-04d5e0170609 09.06.17 17:50 [@ mozilla::net::NeckoParent::GetValidatedOriginAttributes ]\n= bug 1322610\n\n\n(Darkspirit from bug 1322610 comment #9)\n> network.disable.ipc.security;false (default=true)\n> \n> https://gopherproxy.meulie.net/ygrex.ru/h/debian/firefox-security.html\n> > Interprocess security checks (disabled by default per bug 820712):\n> > network.disable.ipc.security=false\n> \n> I just thought that \"disabling\" \"security\" is needed for some old addons or\n> plugins, so I set it to false. \"Security checks\" sounded good.", "id": 12414897, "creation_time": "2017-06-20T16:21:41Z", "author": "jan@ikenmeyer.eu"}, {"count": 5, "text": "(In reply to Darkspirit from comment #4)\n> pdfium\nNo, sorry, I meant pdfjs.", "id": 12414910, "creation_time": "2017-06-20T16:24:40Z", "author": "jan@ikenmeyer.eu"}, {"count": 6, "text": "No user should change network.disable.ipc.security and it will get deleted in the future (post-57). See bug 1322610 comment 6.", "id": 12414919, "creation_time": "2017-06-20T16:27:45Z", "author": "jan@ikenmeyer.eu"}, {"count": 7, "text": "(In reply to Marcus Husar from comment #3)\n> How to reproduce:\n>  * Install Fedo
```
Comment (marco-c) [inline tests/fixtures/bugs.json]: Are you sure you actually need to add more bugs? The first bugs in the list might already have an assignee

Comment (marco-c) [reply tests/fixtures/bugs.json]: Ah you need it because of MINIMUM_ASSIGNMENTS, but you could just overwrite it during the test to be 1.

Comment (probaku1234) [reply tests/fixtures/bugs.json]: I can do that. Is failing the test is related to these added bugs?

Comment (marco-c) [reply tests/fixtures/bugs.json]: Yes, indeed they are passing now :)

File: tests/fixtures/bugs.json [hunk 3]
```diff
@@ -31,4 +31,4 @@
...
...eceive it again on the main thread.  But I'll ask another review from karlt.", "id": 11947159, "creation_time": "2016-12-29T20:05:59Z", "author": "ehsan@mozilla.com"}, {"count": 14, "text": "https://treeherder.mozilla.org/#/jobs?repo=try&revision=608191d559885134ede4dfdb8fbe082b2f481128", "id": 11947160, "creation_time": "2016-12-29T20:06:10Z", "author": "ehsan@mozilla.com"}, {"count": 15, "text": "Yes, backout https://hg.mozilla.org/mozilla-central/rev/90dd139d7578 to test it. This is also the only leak I've seen in reftests, so you can also push the backout along with this patch (or just land this and I'll do it later).", "id": 11947192, "creation_time": "2016-12-29T20:07:44Z", "author": "ahal@mozilla.com"}, {"count": 16, "text": "(In reply to :Ehsan Akhgari from comment #14)\n> https://treeherder.mozilla.org/#/\n> jobs?repo=try&revision=608191d559885134ede4dfdb8fbe082b2f481128\n\nThe patch fixes the leak! \\o/", "id": 11947354, "creation_time": "2016-12-29T21:30:47Z", "author": "ehsan@mozilla.com"}, {"count": 17, "text": "Comment on attachment 8822500\nEnsure that all AudioContext members that need to participate in cycle collection do so\n\nAsyncDecodeWebAudio creates a MediaDe
```
Comment (marco-c) [inline tests/fixtures/bugs.json]: Could you remove the changes from this file altogether?

File: tests/test_assignee.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+from bugbug.models import assignee
+from bugbug.models.assignee import AssigneeModel
+
+
+def test_get_assignee_labels():
+    assignee.MINIMUM_ASSIGNMENTS = 1
+    model = AssigneeModel()
+    classes, _ = model.get_labels()
+    assert len(classes) != 0
+    assert classes[1320039] == "gijskruitbosch+bugs@gmail.com"
```
Comment (marco-c) [inline tests/test_assignee.py]: Could you add a check for another bug too?

Comment (marco-c) [reply tests/test_assignee.py]: At least another bug with an assignee.
And then, assert that a bug with no assignee is not in classes.

File: tests/fixtures/bugs.json (hunk 3 above)
Comment (marco-c) [reply tests/fixtures/bugs.json]: To remove the file, you can simply do:
```
git checkout test-for-assignee
git reset HEAD~1
git checkout -- tests/fixtures/bugs.json
git add tests/test_assignee.py
git commit -m "Add basic test for the Assignee model"
git push --force
```

Comment (probaku1234) [reply tests/fixtures/bugs.json]: hm... I'm not sure I did corretly

Comment (marco-c) [reply tests/fixtures/bugs.json]: No, it's still there :)

Comment (marco-c) [reply tests/fixtures/bugs.json]: Here's another way to do it:
1. Clone this repository in another directory
2. Copy the tests/fixtures/bugs.json from the newly cloned repository to your repository
3. git add tests/fixtures/bugs.json
4. git commit -m "tmp"
5. git rebase -i HEAD~2
6. Select to squash the second commit into the first
7. git push --force origin test-for-assignee

Comment (probaku1234) [reply tests/fixtures/bugs.json]: I did the above. Can you check it?

Comment (marco-c) [reply tests/fixtures/bugs.json]: As you can see, the file is still there :)

Comment (probaku1234) [reply tests/fixtures/bugs.json]: You want to erase the fixtures/bugs.json file?

Comment (marco-c) [reply tests/fixtures/bugs.json]: No, I want the PR not to make any changes to the file.

Comment (probaku1234) [reply tests/fixtures/bugs.json]: Oh...I see....

Comment (marco-c) [reply tests/fixtures/bugs.json]: Let's try yet another thing:
```
cp tests/test_assignee.py tmp.py
git checkout master
git branch -D test-for-assignee
git checkout -b test-for-assignee
mv tmp.py tests/test_assignee.py
git add tests/test_assignee.py
git commit -m "Add basic test for the Assignee model"
git push origin test-for-assignee --force
```

Comment (probaku1234) [reply tests/fixtures/bugs.json]: Ok, I did it. Thanks for help

**Pull request merged**